### PR TITLE
Database schema design

### PR DESCRIPTION
**Pull request by ysoseriouz**
**# Summary:**
1. Init Rails code
2. In folder dbdesign/
- phone_shop_DB.png: design database schema (draw relationship diagram and specify entities - attributes).
- db_requirement.db: implementation note for database migration.
3. In folder db/
- Write migration files to create tables and reference constraints.
- seeds.rb: mock data randomly with code and Faker gem.

Draft database schema diagram for MySQL
phone_shop_DB

### CODE REVIEW

**Review by ducpt1012 — COMMENTED** [review]
1. Naming rules: Database tables use snake_case. Table names are plural. Column names in the database use snake_case, but are generally singular. Ref: https://gist.github.com/iangreenleaf/b206d09c587e8fc6399e#database

2. What the relationship between Model and Brand? How about if I choose the brand Huawei with inputting iPhone 12 ...
3. Does a product has only one image? 
4. If there are many products same properties. Do I have to input all of them?

**Comment by ysoseriouz** [comment]
1. I've fixed the naming.
2. Currently, I let 'model' as free text input for simplification, I think it can be treated the same way as exhaustive list of brands. However, it may take time to get all kind of models, should I call an end to it for now.
3. I've added mechanism to handle many photos.
4. I've added an 'amount' option, I'll note on the wireframes when I implement it.

File: db/seeds.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# This file should contain all the record creation needed to seed the database with its default values.
+# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
+#
+# Examples:
+#
+#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
+#   Character.create(name: 'Luke', movie: movies.first)
+$brand_model = {
```
Comment (dptvn) [inline db/seeds.rb]: use @ instear of $
https://viblo.asia/p/variables-scope-trong-ruby-bWrZnBkmZxw

File: db/migrate/20210706042737_create_inventories.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class CreateInventories < ActiveRecord::Migration[6.1]
+  def change
+    create_table :inventories do |t|
+      t.integer :memory_size, null: false
```
Comment (dptvn) [inline db/migrate/20210706042737_create_inventories.rb]: Should add properties for numeric `unsigned: true`

File: db/migrate/20210706093159_add_status_to_inventories.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToInventories < ActiveRecord::Migration[6.1]
+  def change
+    add_column :inventories, :status, :integer
```
Comment (dptvn) [inline db/migrate/20210706093159_add_status_to_inventories.rb]: try to add it into db/migrate/20210706042737_create_inventories.rb for easy checking

File: db/migrate/20210707135850_create_models.rb [hunk 4]
```diff
@@ -2,8 +2,9 @@ class CreateModels < ActiveRecord::Migration[6.1]
   def change
     create_table :models do |t|
       t.string :name, null: false
+      t.references :brand, null: false, foreign_key: true
 
       t.timestamps
     end
   end
-end
+end
```
Comment (dptvn) [inline db/migrate/20210707135850_create_models.rb]: Should add a new line at the end of the file. Rule not in the paper.

File: db/migrate/20210707143210_create_inventories.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class CreateInventories < ActiveRecord::Migration[6.1]
+  def change
+    create_table :inventories do |t|
+      t.integer :memory_size, null: false, unsigned: true
+      t.integer :manufactoring_year, null: false, unsigned: true
+      t.string :os_version, null: false
+      t.string :color, null: false
+      t.decimal :price, null: false, unsigned: true
+      t.decimal :original_price, unsigned: true
+      t.integer :status, default: 0   # Active
```
Comment (dptvn) [inline db/migrate/20210707143210_create_inventories.rb]: IMO, status filed should add the validation `null: false`

File: db/migrate/20210707135850_create_models.rb (hunk 4 above)
Comment (ysoseriouz) [reply db/migrate/20210707135850_create_models.rb]: Done!

File: db/migrate/20210707143210_create_inventories.rb (hunk 5 above)
Comment (ysoseriouz) [reply db/migrate/20210707143210_create_inventories.rb]: Done!

File: app/models/album.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class Album < ApplicationRecord
+  has_many :photos
+  # belongs_to :photo   # thumbnail_id
```
Comment (dptvn) [inline app/models/album.rb]: belongs_to photo is wrong. Should remove this line if not use

**Review by ducpt1012 — APPROVED** [review]
LGTM

**Review by raksul-sian-izm — APPROVED** [review]
LGTM 🍭

**Comment by raksul-sian-izm** [comment]
What I reviewed 
- Database definitions 
  - erd
  - scheme.rb
- Custom directory's naming way

**Pull request merged**